packages/web/src/lib/myalgowallet-mode.ts

```diff
@@ -89,6 +89,38 @@ export class MyAlgoWalletSession {
 		}
 	}
 
+	/**
+	 * @async
+	 * @description Signs a teal program (https://algorand.github.io/js-algorand-sdk/modules.html#tealSign)
+	 * @param logic Teal program
+	 * @param address Signer Address
+	 * @returns Returns signed teal
+	 * for more info: https://connect.myalgo.com/docs/interactive-examples/TealSign
+	 */
+	async signLogicSignature(logic: string | Uint8Array, address: string): Promise<Uint8Array> {
+		try {
+			return await this.connector.signLogicSig(logic, address)
+		} catch (err) {
+			error(err);
+			throw new Error("Error while signing teal program" + err);
+		}
+	}
+
+	/**
+	 * @description Signs a Logic Signature transaction
+	 * @param transaction algosdk.Transaction object
+	 * @param logicSig Logic Sig Account
+	 * @returns Returns txID and blob object
+	 */
+	signLogicSignatureTxn(transaction: Transaction, logicSig: algosdk.LogicSigAccount): { txID: string, blob: Uint8Array } {
```

Comment:
Let's rename to `createSmartSigTxn` (or maybe there would be even a better name? let's ask others for this one).
Also, let's link: https://developer.algorand.org/docs/get-details/dapps/smart-contracts/smartsigs/modes/#contract-account

Reply:
but we are not creating the txn here, rather we are signing it, so shouldn't be sign

Reply:
As above let's use `Sig` rather than `Signature. Also, in other places we use `Tx` rather than `Txn`.
Otherwise I'm ok here with the `signLogic...` name here.

Reply:
okay updated it to `signLogicSigTx`